client/modules/IDE/components/Editor.js

```diff
@@ -47,23 +49,30 @@ class Editor extends React.Component {
       gutters: ['CodeMirror-lint-markers'],
       keyMap: 'sublime',
       lint: {
-        onUpdateLinting: debounce(2000, (annotations) => {
-          this.props.clearLintMessage();
-          annotations.forEach((x) => {
-            if (x.from.line > -1) {
-              this.props.updateLintMessage(x.severity, (x.from.line + 1), x.message);
+        onUpdateLinting: () => {
+          debounce(2000, (annotations) => {
+            this.props.clearLintMessage();
+            annotations.forEach((x) => {
+              if (x.from.line > -1) {
+                this.props.updateLintMessage(x.severity, (x.from.line + 1), x.message);
+              }
+            });
+            if (this.props.lintMessages.length > 0 && this.props.lintWarning) {
+              this.beep.play();
             }
           });
-          if (this.props.lintMessages.length > 0 && this.props.lintWarning) {
-            this.beep.play();
-          }
-        })
+        }
       }
     });
 
     this._cm.on('change', debounce(200, () => {
```

Comment:
I noticed that typing in the editor is a little sluggish/lags a bit from what you're typing. If you make this debounce a little longer (I tried 1000ms), it's smooth. I think this is because it's too slow rebuilding the iframe over and over again.